dlp/snippets/deid.py

```diff
@@ -2040,6 +2040,188 @@ def deidentify_table_with_multiple_crypto_hash(
+    wrapped_key: bytes = None,
+    alphabet: str = None,
+) -> None:
+    """Uses the Data Loss Prevention API to re-identify sensitive data in a
+    table that was encrypted by Format Preserving Encryption (FPE).
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        table_header: List of strings representing table field names.
+        table_rows: List of rows representing table data.
+        reid_field_names: A list of fields in table to re-identify.
+        key_name: The name of the Cloud KMS key used to encrypt ('wrap') the
+            AES-256 key. Example:
+            key_name = 'projects/YOUR_GCLOUD_PROJECT/locations/YOUR_LOCATION/
+            keyRings/YOUR_KEYRING_NAME/cryptoKeys/YOUR_KEY_NAME'
+        wrapped_key: The decrypted ('wrapped', in bytes) AES-256 key to use. This key
+            should be encrypted using the Cloud KMS key specified by key_name.
+        alphabet: The set of characters to replace sensitive ones with. For
+            more information, see https://cloud.google.com/dlp/docs/reference/
+            rest/v2/projects.deidentifyTemplates#ffxcommonnativealphabet
+    """
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp_v2.DlpServiceClient()
+
+    # Construct the `table`. For more details on the table schema, please see
+    # https://cloud.google.com/dlp/docs/reference/rest/v2/ContentItem#Table
+    headers = [{"name": val} for val in table_header]
+    rows = []
+    for row in table_rows:
+        rows.append({"values": [{"string_value": cell_val} for cell_val in row]})
+    table = {"headers": headers, "rows": rows}
+
+    # Convert table to `item`
+    item = {"table": table}
+
+    # Specify fields to be re-identified/decrypted.
+    reid_field_names = [{"name": _i} for _i in reid_field_names]
+
+    # Construct FPE configuration dictionary
+    crypto_replace_ffx_fpe_config = {
+        "crypto_key": {
+            "kms_wrapped": {"wrapped_key": wrapped_key, "crypto_key_name": key_name}
+        },
+        "common_alphabet": alphabet,
+    }
+
+    # Construct reidentify configuration dictionary
+    reidentify_config = {
+        "record_transformations": {
+            "field_transformations": [
+                {
+                    "primitive_transformation": {
+                        "crypto_replace_ffx_fpe_config": crypto_replace_ffx_fpe_config,
+                    },
+                    "fields": reid_field_names,
+                }
+            ]
+        }
+    }
+
+    # Convert the project id into a full resource id.
+    parent = f"projects/{project}"
+
+    # Call the API.
+    response = dlp.reidentify_content(
+        request={
+            "parent": parent,
+            "reidentify_config": reidentify_config,
+            "item": item,
+        })
+
+    # Print out results.
+    print("Table after re-identification: {}".format(response.item.table))
+
+
+# [END dlp_reidentify_table_fpe]
+
 if __name__ == "__main__":
```

Comment:
Please fix lint:
```
./deid.py:2225:1: C901 'If 2225' is too complex (21)
if __name__ == "__main__":
^
```

Reply:
Hi that is because, max-complexity is set to 20, and we have more than 20 samples now  in deid.py file. There are two solutions:
1. increase max_complexity in noxfile.py
2. create two files for deid related samples. 
Please suggest. Thank you

Reply:
Our sample guidelines recommend using a single file per sample ([see here](https://googlecloudplatform.github.io/samples-style-guide/#one-per-file)). My suggestion is to split it and try to adhere to the above rule in the future. No need to refactor existing samples as there is already a lot of them and that could take long time.

Reply:
Yes, the entire python sample files are maintained that way, We can go ahead and refactor entire dlp folder with 1 snippet/file file, but that is going to take time. Currently lint check is failing because of this complexity, kindly suggest how should I move ahead. Thank you!

Reply:
Please follow @kweinmeister's suggestion (shared in internal chat with @annadata):

> A good starting point, with a low level of effort, would be to analyse the samples for any themes/clusters.  You 
> could then break out the samples into 2+ files, each with a common theme.

I would start with this one and redo any other ones as you extend / modify them. Let me know if you have any further questions.

Reply:
@Avani-Thakker-Crest Please create an issue in backlog with the details of this issue and recommendation to break the samples into their own files.
For all new samples, please create new files.

Reply:
@annadata @rsamborski I created a new file named: deid_table.py (thinking to put all the table related samples in the new file ). I started adding all the table related samples to the new file, but then felt to confirm things first and wanted avoid conflicts with the main branch, hence for now, I have added these two samples (dlp_deidentify_table_fpe and dlp_reidentify_table_fpe) in the new file deid_table.py to get rid of complexity issues, please check.

Reply:
LG from my side with a suggestion to decouple tests as well by putting them in `deid_table_test.py`. If there are any common functions used in `deid_test.py` and `deid_table_test.py` you could put it a helper file.

Reply:
okay got it

Reply:
Added new file deid_table_test.py as per the suggestion. Thank you!